sky/skylet/providers/gcp/node_provider.py

```diff
@@ -239,8 +239,13 @@ def get_order_key(node):
                     reuse_nodes = nodes_matching_launch_config[:count]
                 else:
                     nodes_all = resource._list_instances(filters, STOPPED_STATUS)
+                    nodes_matching_launch_config_ids = {
```

Comment:
nit: prefer to write `set(n.id for n ....)`? `{}` on first glance can be invoke "dict".